### DISCUSSION TITLE
Object reference not set to an instance of an object

### DISCUSSION THREAD

**Discussion by ihorkolucky** [abuzuhri/Amazon-SP-API-CSharp#41] [Q&A] [answered]
Hello,

I try to use the package to get orders, but it throws an error: Object reference not set to an instance of an object

on line of code:

AmazonConnection amazonConnection = new AmazonConnection(credential);

I passed all my credentials (trimmed for security reason) below: 

AmazonCredential credential = new AmazonCredential()
            {
                AccessKey = "AKIATG.....................",
                SecretKey = "iQdcAy......................",
                RoleArn = "arn:aws:iam::220871........:role/IAM_Role_SellingPartnerAPI",
                ClientId = "amzn1.application-oa2-client.................",
                ClientSecret = "83bba86f1.............",
                RefreshToken = "Atzr......................."
            };

Stack trace:
   at FikaAmazonAPI.Services.RequestService..ctor(AmazonCredential amazonCredential)
   at FikaAmazonAPI.Services.OrderService..ctor(AmazonCredential amazonCredential)
   at FikaAmazonAPI.AmazonConnection.Init(AmazonCredential Credentials)
   at FikaAmazonAPI.AmazonConnection.Authenticate(AmazonCredential Credentials)
   at FikaAmazonAPI.AmazonConnection..ctor(AmazonCredential Credentials)
   at GetOrders.Program.Main(String[] args)

Could you please advise?

Thank you!

**Comment by wheelersf** [comment, marked as answer]
Hi Ihorkolucky,

Should be and easy fix.  Just add market reference to the the credentials object:

AmazonConnection amazonConnection = new AmazonConnection(new AmazonCredential()
                {
                    AccessKey = ";alsdkjf;lsd",
                    SecretKey = "e/;alsdkjf;alsdjk",
                    RoleArn = "arn:aws:iam::;lasdjkf;aldsjk:role/;lasdjkf;aldkj",
                    ClientId = "amzn1.application-oa2-client.l;asjdf;ladsjk",
                    ClientSecret = ";alsdjf;ldsjkf",
                    RefreshToken = "Atzr|;alskd;lasdkj",
                    MarketPlace = MarketPlace.US
                });

**Reply by ihorkolucky** [reply]
It works, thank you!

**Reply by abuzuhri** [reply]
You welcome
